### DISCUSSION TITLE
on_build 如何使用？没有找到参考

### DISCUSSION THREAD

**Discussion by q962** [xmake-io/xmake#4361] [General]
不知道源文件中是否存在用例，搜索没有找到相关内容。

找到一个 `on_build("build.target")`，但是根本无从下手。

在 on_build 中，我可以通过 compiler.compile 编译文件
但是提示 `unknown source kind!`

编译出来的 .o xmake 会接管吗？还是需要自己编写 on_link?

**Comment by q962** [comment]
xmake v2.8.2+HEAD.57212c6

**Reply by waruqi** [reply]
https://xmake.io/#/zh-cn/manual/project_target?id=targeton_build

on_build 是重写整个 build，包括所有文件，不仅仅只是单个文件的编译

单个文件的构建走 xxx_buildcmd_file/xxx_build_file 系接口

https://github.com/xmake-io/xmake/blob/2cb60351fb3730a0a240f4393bd5419d2ef0b143/xmake/rules/lex_yacc/yacc/xmake.lua#L29

**Reply by q962** [reply]
我目前在为 target 编写 on_build

**Reply by q962** [reply]
难度超过我的想象，我放弃了。我还是在 on_load 里做文章吧

**Comment by waruqi** [comment]
> 难度超过我的想象，我放弃了。我还是在 on_load 里做文章吧

除非要整个重写 target 构建，我自己都不用，看下现有提供的 rules ，建议使用 buildcmd_file
